DNN Platform/Library/Common/Globals.cs

```diff
@@ -3444,6 +3444,13 @@ public static bool IsHostTab(int tabId)
             return PortalSecurity.Instance.InputFilter(strSQL, PortalSecurity.FilterFlag.NoSQL);
         }
 
+        /// <summary>Gets an <see cref="IServiceProvider"/> for the current request, or the global provider if not available.</summary>
+        /// <returns>An <see cref="IServiceProvider"/> instance.</returns>
+        public static IServiceProvider GetCurrentServiceProvider()

```

Comment:
I made this helper method public, since I needed it in the following:

https://github.com/dnnsoftware/Dnn.Platform/blob/d302d0755d44f04217fbfdf8368ac62936aa4ebc/DNN%20Platform/Modules/ResourceManager/Services/Dto/FolderPermissions.cs#L72-L76

Is this OK?

The only problem with calling `GetCurrentServiceProvider` is that it falls back to the global service provider if the HttpContext is not available. Since IPermissionService is a scoped service, it'll create a scoped instance in the root provider that won't be disposed. This is currently not an issue (since we don't implement IDisposable), but maybe it'll be in the future. 🤷‍♂️

Reply:
Alternativly I could add an `[assembly: InternalsVisibleTo("Dnn.Modules.ResourceManager")]`... But this sounds hacky 😅

Reply:
I would update `Permissions.LoadPermissionDefinitions` to take an `IPermissionService` instance.

Reply:
I refactored LoadPermissionDefinitions; but for this I to add `IPermissionService` as a parameter in a constructor, which I don't know if I like.

This only affects `Dnn.Modules.ResourceManager`. Can we make breaking changes in here or is this still the core of DNN?

Reply:
We can make breaking changes in Resource Manager

Reply:
I agree, we don't publish a Nuget for this one and it is a module that could in some scenarios not even be installed...